### ISSUE TITLE
Question, open file dialog with some additional options

### ISSUE THREAD

**Issue by drmcnelson** [JamesBremner/windex#32]
Hi,

Can you show or tell me how to create a dialog for opening a new file, and include some extra checkboxes?

Thank you

**Comment by JamesBremner** [comment]
Do you want more than provided by filebox?  https://jamesbremner.github.io/windex/classwex_1_1filebox.html

If so, can you provide more details?

**Comment by drmcnelson** [comment]
To be specific, this is for the Save dialog for the spectrometer.

The idea is to offer a way to choose to save only the most recent or some range of ranges or all of the frames available.

Aside this can have different meaning depending on how the frames were collected.  In a triggered burst, it is a kinetic series that documents the time evolution of some phenomenon.   For simple clocked operation, it is like a history record, and most often it is only the last that needs to be saved.

Alternatively I can just remember which state the data belongs to and save accordingly.

So, without being able to do that in the widget, I am still okay.   In other words it is not a high priority,  if its there already or you can say how to do it off the top of your head, great.  Otherwise I can leave it for another time.

**Comment by drmcnelson** [comment]
I think I would be ok with closing this, I can manage without it for the time being.  Sorry for taking up time on it.

**Comment by JamesBremner** [comment]
Those are not the details I was asking for.

So, let me propose:

A dialog box with

- a button that uses filebox to select a file to save to.  When file selected, a label is filled with the selected file path and the SAVE button is enabled
- a SAVE button, initially disabled.
- Checkboxes, etc
- A CANCEL button, to abandon save and close dialog

Use case:

- Select a file to save
- Select checkboxes etc to configure save process
- Click save to the selected file, respecting the choices in the checkboxes etc
-

**Comment by drmcnelson** [comment]
That's wonderfully useful, thank you.   

 As I recall, from many years ago,  file dialogs have some compatibility to save what directory they are in for the next time they are used, and to allow specific file type suffixes, yes?

**Comment by JamesBremner** [comment]
Yes.  take a look at the filebox docs for those extras https://jamesbremner.github.io/windex/classwex_1_1filebox.html

**Comment by drmcnelson** [comment]
The links is a "404"  not found

**Comment by JamesBremner** [comment]
You are right, those are old links.

Are you still interested in seeing the demo?  I can probably recover it without too much trouble, tomorrow

**Comment by drmcnelson** [comment]
I think I have the file extension thing worked out.   The rest has become less important to the task at hand, I worked out a protocol that makes the options unnecessary.
